### PR TITLE
fix(animations): ensure void => * animations are triggered when an expression is omitted

### LINKED ISSUE #9327: Animations states without an expression are ignored for `void => something`

### PR DESCRIPTION
**Pull request by matsko**
Closes #9327

### CODE REVIEW

File: modules/@angular/compiler/src/view_compiler/property_binder.ts [hunk 1]
```diff
@@ -31,12 +31,12 @@ function createCurrValueExpr(exprIndex: number): o.ReadVarExpr {
 function bind(
     view: CompileView, currValExpr: o.ReadVarExpr, fieldExpr: o.ReadPropExpr,
     parsedExpression: cdAst.AST, context: o.Expression, actions: o.Statement[],
-    method: CompileMethod) {
+    method: CompileMethod, allowBlankExpression: boolean) {
```
Comment (tbosch) [inline modules/@angular/compiler/src/view_compiler/property_binder.ts]: This seems weird. The purpose of the flag `allowBlankExpression` is to do nothing. Could we then just not call this method at all?

File: modules/@angular/compiler/src/view_compiler/property_binder.ts [hunk 2]
```diff
@@ -89,6 +89,7 @@ function bindAndWriteToRenderer(
     var fieldExpr = createBindFieldExpr(bindingIndex);
     var currValExpr = createCurrValueExpr(bindingIndex);
     var renderMethod: string;
+    var allowBlankExpression = false;
```
Comment (tbosch) [inline modules/@angular/compiler/src/view_compiler/property_binder.ts]: Create a new function for each of these `case`s of the switch that also contain (or not contain for animations) the call to `bind` and use them here. E.g. a `bindRenderAttribute` / `bindRenderProperty` / `bindRenderClass` / `bindRenderStyle` / `bindRenderAnimation`.

Comment (tbosch) [inline modules/@angular/compiler/src/view_compiler/property_binder.ts]: Then you don't need the flag `allowBlankExpression` any more...

File: modules/@angular/core/test/animation/animation_integration_spec.ts [hunk 3]
```diff
@@ -142,6 +142,55 @@ function declareTests({useJit}: {useJit: boolean}) {
...
+                     tick();
+
+                     var cmp = fixture.debugElement.componentInstance;
+                     cmp.exp = true;
+                     fixture.detectChanges();
+                     flushMicrotasks();
+
+                     expect(driver.log.length).toEqual(1);
+
+                     var animation = driver.log[0];
```
Comment (tbosch) [inline modules/@angular/core/test/animation/animation_integration_spec.ts]: Could you assert all of these properties at once with an `expect(driver.log[0]).toEqual({duration: 500, ...})`? Would make the code shorter and easier to read.

File: modules/@angular/core/test/animation/animation_integration_spec.ts [hunk 4]
```diff
@@ -142,6 +142,55 @@ function declareTests({useJit}: {useJit: boolean}) {
...
+                     expect(startingStyles).toEqual({'opacity': '0'});
+
+                     var keyframes = animation['keyframeLookup'];
+                     expect(keyframes[1]).toEqual([1, {'opacity': '1'}]);
+
+                     cmp.exp = false;
+                     fixture.detectChanges();
+                     flushMicrotasks();
+
+                     animation = driver.log[1];
```
Comment (tbosch) [inline modules/@angular/core/test/animation/animation_integration_spec.ts]: See above, use an `expect(driver.log[1]).toEqual(...)`

**Comment by IgorMinar** [comment]
@matsko can you fix this up and merge it please?

**Comment by vicb** [comment]
Status ?

**Pull request closed without merging**